VDA-XP-Reward/contracts/VDAXPReward.sol

```diff
@@ -0,0 +1,277 @@
+        }
+    }
+
+    /**
+     * @notice Check the proofs and return the total amount of XP
+     * @param didAddress - DID address
+     * @param infos - Array of claim requests
+     * @return uint - Total amount of requested claim
+     */
+    function _validateClaimSignature(address didAddress, ClaimInfo[] calldata infos) internal virtual returns(uint) {
+        uint totalXP;
+
+        bytes memory rawMsg;
+        bytes32 dataHash;
+        address proofSigner;
+
+        uint infoLen = infos.length;
+        DateTime._DateTime memory curTime = DateTime.parseTimestamp(block.timestamp);
+
+        for (uint i; i < infoLen;) {
+            if (infos[i].xp == 0) {
+                revert InvalidXP(didAddress, infos[i].typeId, infos[i].xp);
+            }
+            _validateProofTime(didAddress, curTime, infos[i]);
+
+            rawMsg = abi.encodePacked(didAddress, infos[i].typeId, infos[i].issueYear, infos[i].issueMonth);
+            
+            // Check whether `typeId` is claimed in this month
+            if (isClaimedTypeId[rawMsg]) {
+                revert DuplicatedRequest(didAddress, infos[i].typeId, infos[i].xp, infos[i].signature);
+            }
+            isClaimedTypeId[rawMsg] = true;
+
+            if (bytes(infos[i].uniqueId).length != 0) {
+                rawMsg = abi.encodePacked(infos[i].typeId, infos[i].uniqueId);
+                // Check whether `uniqueId` is claimed before
+                if (isClaimedUniqueId[rawMsg]) {
+                    revert DuplicatedUniqueId(didAddress, infos[i].typeId, infos[i].uniqueId, infos[i].xp);
+                }
+                isClaimedUniqueId[rawMsg] = true;
+
+                rawMsg = abi.encodePacked(didAddress, rawMsg, infos[i].issueYear, infos[i].issueMonth, infos[i].xp);
+                
+            } else {
+                rawMsg = abi.encodePacked(rawMsg, infos[i].xp);
+                // Check `Proof` is claimed
+                if (isClaimedSignature[infos[i].signature]) {
+                    revert DuplicatedRequest(didAddress, infos[i].typeId, infos[i].xp, infos[i].signature);
+                }
+                isClaimedSignature[infos[i].signature] = true;
+
+            }
+
+            dataHash = keccak256(rawMsg);
+            proofSigner = ECDSAUpgradeable.recover(dataHash, infos[i].signature);
+
+            _validateProofSigner(didAddress, proofSigner, infos[i]);
+            
+            totalXP = totalXP + infos[i].xp;
+            unchecked {++i;}
+        }
+
+        return totalXP;
+    }
+
+    /**
+     * @dev See {IVDAXPReward}
+     */
+    function claimXPReward(
+        address didAddress, 
+        address to,
+        ClaimInfo[] calldata claims,
+        bytes calldata requestSignature,
+        bytes calldata requestProof
+    ) external virtual override {
+        if (conversionRate == 0) {
+            revert InvalidConversionRate();
+        }
+        if (!didRegistryContract.isRegistered(didAddress)) {
```

Comment:
Let's remove this.

In the future we will support DID's registered on other blockchains, so this will break.

Reply:
Removed.